L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -1,110 +1,163 @@
 #include "L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorSuperTriggerCellImpl.h"
-#include "DataFormats/ForwardDetId/interface/HGCalTriggerDetId.h"
-
-#include <unordered_map>
 
 HGCalConcentratorSuperTriggerCellImpl::HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf)
-    : stcSize_(conf.getParameter<std::vector<unsigned> >("stcSize")) {
-  if (stcSize_.size() != kNLayers_) {
-    throw cms::Exception("HGCTriggerParameterError")
-        << "Inconsistent size of super trigger cell size vector" << stcSize_.size();
-  }
-  for (auto stc : stcSize_) {
-    if (stc != kSTCsizeFine_ && stc != kSTCsizeCoarse_) {
-      throw cms::Exception("HGCTriggerParameterError")
-          << "Super Trigger Cell should be of size " << kSTCsizeFine_ << " or " << kSTCsizeCoarse_;
-    }
+    : fixedDataSizePerHGCROC_(conf.getParameter<bool>("fixedDataSizePerHGCROC")),
+      coarseTCmapping_(std::vector<unsigned>{HGCalCoarseTriggerCellMapping::kCTCsizeVeryFine_,
+                                             HGCalCoarseTriggerCellMapping::kCTCsizeVeryFine_,
+                                             HGCalCoarseTriggerCellMapping::kCTCsizeVeryFine_,
+                                             HGCalCoarseTriggerCellMapping::kCTCsizeVeryFine_}),
+      superTCmapping_(conf.getParameter<std::vector<unsigned>>("stcSize")) {
+  std::string energyType(conf.getParameter<string>("type_energy_division"));
+
+  if (energyType == "superTriggerCell") {
+    energyDivisionType_ = superTriggerCell;
+  } else if (energyType == "oneBitFraction") {
+    energyDivisionType_ = oneBitFraction;
+
+    oneBitFractionThreshold_ = conf.getParameter<double>("oneBitFractionThreshold");
+    oneBitFractionLowValue_ = conf.getParameter<double>("oneBitFractionLowValue");
+    oneBitFractionHighValue_ = conf.getParameter<double>("oneBitFractionHighValue");
+
+  } else if (energyType == "equalShare") {
+    energyDivisionType_ = equalShare;
+
+  } else {
+    energyDivisionType_ = superTriggerCell;
   }
 }
 
-const std::map<int, int> HGCalConcentratorSuperTriggerCellImpl::kSplit_ = {{kSTCsizeFine_, kSplit_v8_Fine_},
-                                                                           {kSTCsizeCoarse_, kSplit_v8_Coarse_}};
+void HGCalConcentratorSuperTriggerCellImpl::createAllTriggerCells(
+    std::unordered_map<unsigned, SuperTriggerCell>& STCs, std::vector<l1t::HGCalTriggerCell>& trigCellVecOutput) const {
+  for (auto& s : STCs) {
```

Comment:
could this be const ref?

Reply:
The STC does need to be modified in this line 
`s.second.addToFractionSum(tc_fraction);`